### ISSUE TITLE
UMP - UnityException: Internal_CreateGameObject called off mainthread in onConsentFormDismissed

### ISSUE THREAD

**Issue by dginovker** [googleads/googleads-mobile-unity#2985]
### Step 0: Are you in the right place?

  * For general technical questions, or help with project-specific issues like setting up ads in
   your app, reach out to our support team on the
   [Developer Forum](https://groups.google.com/forum/#!categories/google-admob-ads-sdk/game-engines).
  * For assistance with your AdMob account, reach out to
   [AdMob Support](https://support.google.com/admob/?hl=en#topic=7383088).
  * For feedback on [our documentation](https://developers.google.com/admob/unity/start),
   send  your feedback by pressing the **Send Feedback** button at the top right of the
    documentation page you are on.
  * For issues related to __the code in this repository__, continue filing this GitHub issue.

  * Once you've read this section and determined that your issue is appropriate for
    this repository, **please delete this section**.

### [REQUIRED] Step 1: Describe your environment

  * Unity version: 2021.3.9f1
  * Google Mobile Ads Unity plugin version: 8.5.2
  * Platform: Android, iOS (iOS, Android, Unity Editor)
  * Platform OS version: iOS 17, Android 14
  * Any specific devices issue occurs on: All
  * Mediation ad networks used, and their versions: IronSource 7.5.3, Applovin 5.9.0

#### Steps to reproduce:

* Have the below code
* Open the app on Android
* Read ADB logs

#### Relevant Code:

```
[NonSerialized] public string UMPDebugModeDeviceIdentifiers = "[\"ABCDEFABCDEFABCDEFABCDEFABCDEFAB\"]"; // Search for "Use new ConsentDebugSettings" on ADB or "To enable debug mode" on iOS

void Start {
    var debugSettings = new ConsentDebugSettings
    {
        DebugGeography = (DebugGeography)PlayerPrefs.GetInt("DebugUMPGeography", (int)DebugGeography.Disabled),
        TestDeviceHashedIds = Newtonsoft.Json.JsonConvert.DeserializeObject<List<string>>(UMPDebugModeDeviceIdentifiers)
    };
    Debug.Log($"UMP Consent Debug Settings: DebugGeography: {debugSettings.DebugGeography}, TestDeviceHashedIDs: {string.Join(", ", debugSettings.TestDeviceHashedIds)}");
    ConsentRequestParameters request = new ConsentRequestParameters
    {
        ConsentDebugSettings = debugSettings,
    };

    // Check the current consent information status.
    ConsentInformation.Update(request, OnConsentInfoUpdated);
}

void OnConsentInfoUpdated(FormError consentError)
{
    if (consentError != null)
    {
        // Handle the error.
        UnityEngine.Debug.LogError($"UMP Consent Error code: {consentError.ErrorCode}, message: {consentError.Message}");
        return;
    }

    // If the error is null, the consent information state was updated.
    // You are now ready to check if a form is available.
    ConsentForm.LoadAndShowConsentFormIfRequired((FormError formError) =>
    {
        if (formError != null)
        {
            // Consent gathering failed.
            UnityEngine.Debug.LogError($"Failed to load consent form. Error code: {formError.ErrorCode}, message: {formError.Message}");
            return;
        }

        // Consent has been gathered.
        if (ConsentInformation.CanRequestAds())
        {
            MobileAds.Initialize((InitializationStatus initstatus) =>
            {
                string AdapterStatusToString(AdapterStatus status)
                {
                    return
                        $"Description: {status.Description} - Init state: {status.InitializationState} - Latency: {status.Latency}";
                }
                foreach(var key in initstatus.getAdapterStatusMap().Keys) {
                    Debug.Log($"UMP initStatus key {key} : {AdapterStatusToString(initstatus.getAdapterStatusMap()[key])}");
                }
            });
        }
    });
}
```

#### Logs:

```
10-25 09:35:11.113  2689  2937 E Unity   : UnityException: Internal_CreateGameObject can only be called from the main thread.
10-25 09:35:11.113  2689  2937 E Unity   : Constructors and field initializers will be executed from the loading thread when loading a scene.
10-25 09:35:11.113  2689  2937 E Unity   : Don't use this function in the constructor or field initializers, instead move initialization code to the Awake or Start function.
10-25 09:35:11.113  2689  2937 E Unity   :   at GoogleMobileAds.Common.MobileAdsEventExecutor.Initialize () [0x00000] in <00000000000000000000000000000000>:0 
10-25 09:35:11.113  2689  2937 E Unity   :   at GoogleMobileAds.Ump.Android.ConsentFormClient.onConsentFormDismissed (UnityEngine.AndroidJavaObject error) [0x00000] in <00000000000000000000000000000000>:0 
10-25 09:35:11.113  2689  2937 E Unity   :   at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <00000000000000000000000000000000>:0 
10-25 09:35:11.113  2689  2937 E Unity   :   at UnityEngine.AndroidJavaProxy.Invoke (System.String methodName, System.Object[] args) [0x00000] in <00000000000000000000000000000000>:0 
10-25 09:35:11.113  2689  2937 E Unity   :   at UnityEngine._Android
```

Happens 100% of the time on both devices tested

**Comment by dginovker** [comment]
Looks like the issue is right here: https://github.com/googleads/googleads-mobile-unity/blob/main/source/plugin/Assets/GoogleMobileAds/Common/MobileAdsEventExecutor.cs#L39C34-L39C44

Initialize() is being called off the main thread, but is trying to initialize a Gameobject.

**Comment by NVentimiglia** [comment]
@dginovker 

Thanks for the review. I have filed the bug and will get it fixed b/307745296.

Is it possible to call `MobileAdsEventExecutor.Initialize()` in Start before running the UMP code as a work around?
